signalfx-forwarder-app/bin/eventstosfx.py

```diff
@@ -0,0 +1,171 @@
+# pylint: skip-file
+import gzip
+import json
+import os
+import sys
+from io import BytesIO
+
+current_path = os.path.dirname(__file__)  # pylint: disable=invalid-name
+sys.path.append(os.path.join(current_path, "libs"))
+sys.path.append(os.path.join(current_path, "libs", "sfxlib"))
+
+from splunklib.searchcommands import (  # isort:skip pylint: disable=import-error
+    Configuration,
+    EventingCommand,
+    Option,
+    dispatch,
+    validators,
+)
+
+import requests  # isort:skip
+
+
+@Configuration()
+class EventToSFXCommand(EventingCommand):
+    """
+    ## Syntax
+
+    <command> | eventstosfx
+
+    ## Description
+
+    One or more datapoints are generated for each input event's field(s) of the
+    form `gauge_*`, `counter_*` or `cumulative_counter_*`.  The metric name in
+    SignalFx will be the `*` part of the field name.  Any additional fields on
+    the event will be attached as dimensions to the generated datapoints.
+
+    """
+
+    access_token = None
+    debug = Option(validate=validators.Boolean(), default=False)
+    dry_run = Option(validate=validators.Boolean(), default=False)
+    ingest_url = Option(validate=validators.Match("https://.*", r"^https://.*"))
+    ev_endpoint = Option(default="/v2/event")
+
+    SPLUNK_PASSWORD_REALM = "realm"
+    SPLUNK_PASSWORD_USER_NAME = "username"
+    SPLUNK_PASSWORD_CLEAR_PASSWORD = "clear_password"
+    SPLUNK_KV_STORE_SFX_CONFIG_COLLECTION_NAME = "sfx_ingest_config"
+    SPLUNK_SFX_CONFIG_INGEST_URL_KEY = "ingest_url"
+    SPLUNK_PASSWORDS_STORAGE_SFX_ACCESS_TOKEN_REALM = "sfx_ingest_command"
+    SPLUNK_PASSWORDS_STORAGE_SFX_ACCESS_TOKEN_USER_NAME = "access_token"
+
+    def ensure_default_config(self):
```

Comment:
nit, this method doesn't appear to do what the name suggests.  Maybe something like `ensure_connection_info()`?

Reply:
I copied the original naming convention from whoever put tosfx.py in